pkg/ctl/install/install.go

```diff
@@ -0,0 +1,444 @@
+	cloneDir, err := ioutil.TempDir(os.TempDir(), "eksctl-install-flux-clone")
+	if err != nil {
+		return "", fmt.Errorf("cannot create temporary directory: %s", err)
+	}
+	cloneCtx, cloneCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer cloneCtxCancel()
+	args := []string{"clone", "-b", fi.opts.templateParams.GitBranch, fi.opts.templateParams.GitURL, cloneDir}
+	err = runGitCmd(cloneCtx, cloneDir, args...)
+	return cloneDir, err
+}
+
+func (fi *fluxInstaller) addCommitAndPushFluxManifests(ctx context.Context, cloneDir string) error {
+	// git add
+	addCtx, addCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer addCtxCancel()
+	if err := runGitCmd(addCtx, cloneDir, "add", "--", fi.opts.gitFluxPath); err != nil {
+		return err
+	}
+
+	// Confirm there is something to commit, otherwise move on
+	diffCtx, diffCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer diffCtxCancel()
+	if err := runGitCmd(diffCtx, cloneDir, "diff", "--cached", "--quiet", "--", fi.opts.gitFluxPath); err == nil {
+		logger.Info("Nothing to commit (the repository contained identical manifests), moving on")
+		return nil
+	} else if _, ok := err.(*exec.ExitError); !ok {
+		return err
+	}
+
+	// git commit
+	commitCtx, commitCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer commitCtxCancel()
+	args := []string{"commit",
+		"-m", "Add Initial Flux configuration",
+		fmt.Sprintf("--author=%s <%s>", fi.opts.templateParams.GitUser, fi.opts.templateParams.GitEmail),
+	}
+	if err := runGitCmd(commitCtx, cloneDir, args...); err != nil {
+		return err
+	}
+
+	// git push
+	pushCtx, pushCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer pushCtxCancel()
+	return runGitCmd(pushCtx, cloneDir, "push")
+}
+
+func (fi *fluxInstaller) applyManifests(manifestsMap map[string][]byte) error {
+	// TODO: initialise the client elsewhere so that a mock client can easily be dependency-injected for testing purposes.
+	client, err := kubernetes.NewRawClient(fi.k8sClientSet, fi.k8sRestConfig)
+	if err != nil {
+		return err
+	}
+
+	// If the namespace needs to be created, do it first before any other
+	// resource which should potentially be created within this namespace.
+	// Otherwise, creation of these resources will fail.
+	if namespace, ok := manifestsMap[namespaceFileName]; ok {
+		if err := createOrReplace(namespace, client); err != nil {
+			return err
+		}
+		delete(manifestsMap, namespaceFileName)
+	}
+	manifests := kubernetes.ConcatManifestValues(manifestsMap)
+	return createOrReplace(manifests, client)
+}
+
+func createOrReplace(manifest []byte, client *kubernetes.RawClient) error {
+	objects, err := kubernetes.NewRawExtensions(manifest)
+	if err != nil {
+		return err
+	}
+	for _, object := range objects {
+		if err := createOrReplaceObject(object, client); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createOrReplaceObject(object runtime.RawExtension, client *kubernetes.RawClient) error {
```

Comment:
Should I move this to some utility package? It feels like something which could be re-used.